aten/src/ATen/functorch/BatchRulesReduceOps.cpp

```diff
@@ -133,7 +156,39 @@ void boxed_reduction_batch_rule(const c10::OperatorHandle& op, torch::jit::Stack
     new_dims.push_back(getPhysicalDim(self, self_bdim.has_value(), dim));
   }
   bool is_scalar_case = logical_dim == 0 && dims.size() == 1 && is_allowed_dim_on_scalar_tensor(dims[0]);
+  c10::optional<bool> maybe_keepdim;
   if (is_scalar_case) {
+    // NOTE: [boxed_reduction_batch_rule scalar tensor handling]
+    // Reduction operations in PyTorch have an edge case where they allow
+    // dim=0 and dim=-1 if the tensor has shape [].
+    //
+    // This can come up if we do something like
+    // vmap(lambda x: x.sum(0))(torch.tensor([10.])),
+    //
+    // In order to handle this edge case, we unsqueeze a dimension on the Tensor,
+    // run the operation (with dim=1 instead), and then process the output tensor.
```

Comment:
What do you mean by "treating this like a no op"?

Reply:
Oh, you're wondering why we need to perform op to go from [B, 1] -> [B, 1], because it looks like a no-op.

It is still important that we run the op, because the op does something different depending on what the value inside the Tensor is.

Some cases:
- Let's say we have x = tensor(1.). Then x.sum(0) returns 1., which is indeed a no-op.
- However, consider x = tensor(float('nan')). x.sum(0) returns float('nan'), but x.nansum(0) returns 0.